File: backend-nestjs/apps/ai/src/knowledge/entities/knowledge-base.entity.ts
import { Entity, Column, PrimaryGeneratedColumn } from 'typeorm';

@Entity('knowledge_bases')
export class KnowledgeBaseEntity {
  @PrimaryGeneratedColumn({ type: 'bigint' })
  id: string;

  @Column({ name: 'tenant_id', type: 'bigint' })
  tenantId: string;

  @Column({ name: 'org_id', type: 'bigint' })
  orgId: string;

  @Column({ name: 'created_by', type: 'bigint' })
  createdBy: string;

  @Column()
  name: string;

  @Column({ nullable: true })
  description: string;
}

File: backend-java/cdm-ai/src/main/java/com/cdm/ai/entity/KnowledgeBaseEntity.java
package com.cdm.ai.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.cdm.common.domain.BaseEntity;
import com.cdm.ai.vo.KnowledgeBaseVo;
import lombok.Getter;
import lombok.Setter;
import lombok.NoArgsConstructor;

@TableName("knowledge_bases")
@Getter
@Setter
@NoArgsConstructor
public class KnowledgeBaseEntity extends BaseEntity {

    private Long tenantId;
    private Long orgId;
    private Long createdBy;
    private String name;
    private String description;

    public static KnowledgeBaseVo toVo(KnowledgeBaseEntity entity) {
        if (entity == null) return null;
        return KnowledgeBaseVo.builder()
                .id(entity.getId())
                .tenantId(entity.getTenantId())
                .orgId(entity.getOrgId())
                .createdBy(entity.getCreatedBy())
                .name(entity.getName())
                .description(entity.getDescription())
                .createdAt(entity.getCreatedAt())
                .build();
    }
}
